ha/core/event_manager/event_manager.py

```diff
@@ -320,16 +322,10 @@ def subscribe(self, component: SUBSCRIPTION_LIST, events: List[SubscribeEvent])
                 subscription_key = EVENT_MANAGER_KEYS.SUBSCRIPTION_KEY.value.replace("<component_id>", component)
                 if event.functional_types:
                     for func_type in event.functional_types:
-                        resource = event.resource_type + HA_DELIM + func_type       # Example: resource = node>server
-                        self._store_component_key(subscription_key, resource, event.states)
-                        self._store_event_key(resource, event.states, comp=component)
+                        self._store_component_key(subscription_key, event.resource_type, func_type, event.states)
+                        self._store_event_key(event.resource_type, func_type, event.states, comp=component)
                         for state in event.states:
-                            self._monitor_rule.add_rule(resource, state, self._default_action)
-                else:
-                    self._store_component_key(subscription_key, event.resource_type, event.states)
```

Comment:
Is this else not needed in the final patch ?

Reply:
event.functional_types will have values always. So removed if-else.